test/integration/patients/patient/dashboard.js

```diff
@@ -376,198 +402,258 @@ context('patient dashboard page', function() {
 
     cy
       .url()
-      .should('contain', 'patient-action/1/form/1');
+      .should('contain', `patient-action/${ testAction.id }/form/${ testForm.id }`);
   });
 
   specify('add action and flow', function() {
+    const currentClinican = getCurrentClinician({
+      relationships: {
+        role: getRelationship(roleEmployee),
+      },
+    });
+
+    const testPrograms = [
+      getProgram({
+        attributes: {
+          name: 'Two Actions, One Published, One Flow',
+          published_at: testTs(),
+          archived_at: null,
+        },
+        relationships: {
+          'program-flows': getRelationship([{ id: '4' }], 'flows'),
+          'program-actions': getRelationship(
+            [{ id: '1' }, { id: '4' }, { id: '5' }, { id: '6' }],
```

Comment:
all of the ids and the ones below should be able to be done by using only `getProgram`, `getProgramAction` and `getProgramFlow` except that you'll need `uuid`s to setup the circular ids.. should be like on patient-flow or program-flow tests as well.

Reply:
I've got a solution for this. But it feels overly complicated. Open to any better ideas.